### DISCUSSION TITLE
postUpgradeTasks command is not being executed - Self hosted Renovate

### DISCUSSION THREAD

**Discussion by gjorgi93** [renovatebot/renovate#9451] [Request Help]
I'm using self-hosted instance of Renovate, my goal is to add postUpgradeTask command to be executed after npm install command.  My renovate.json file is:
 
```{
  
"$schema": "https://docs.renovatebot.com/renovate-schema.json",  
  "commitMessagePrefix": "deps: ",  
  "extends": [
    "config:base"
  ],
  "ignorePresets": [":prHourlyLimit2"],
  "recreateClosed": true,
  "prConcurrentLimit": 0,
  "separateMajorMinor": false,
  "ignoreDeps": ["go.opentelemetry.io/otel", "github.com/Azure/azure-service-bus-go"],
  "allowedPostUpgradeCommands": ["^npm ci --ignore-scripts$", "^npx ng update"],
  "trustLevel": "high",
  "semanticCommits": true,
  "allowPostUpgradeCommandTemplating": true,
  "packageRules": [
    {
      "matchPackagePatterns": ["^@angular", "^@types", "^tslint"],
      "rangeStrategy": "pin",
      "postUpgradeTasks": {
        "commands": [
          "npm ci --ignore-scripts",
          "npx ng update {{{depName}}} --from={{{currentVersion}}} --to={{{newVersion}}} --migrateOnly --allowDirty --force"
        ],
        "fileFilters": ["**/**"]
      }
    }
  ]
}
```

If i'm missing something in my configuration, please let me know. Thanks

**Comment by rarkins** [comment]
Admin options like `trustLevel` and `allowedPostUpgradeCommands` just be in the bot config, not the repo `renovate.json`. i.e. the `config.js` which the bot reads

**Reply by gjorgi93** [reply]
Thanks @rarkins , i moved those options into the config.js file and now works as expected.
After execution of the command i got the following artifact err:
```
       "artifactErrors": [
         {
           "lockFile": "code/js/frontend/package.json",
           "stderr": "Command failed: npx ng update @angular/animations --from= --to=11.2.6 --migrateOnly --allowDirty --force\nnpx: installed 1 in 1.881s\ncommand not found: ng\n"
         }
```

on each angular package.

**Reply by rarkins** [reply]
There is an example version similar to what you need in https://docs.renovatebot.com/self-hosted-configuration/#allowpostupgradecommandtemplating

**Reply by gjorgi93** [reply]
Does `npm ci --ignore-scripts` and `npx ng update` commands have to be added elsewhere? i don't have them in my package.json file. I followed the same example from the documentation, still having the same artifact error that i posted above, saying that ng command is not found...

**Reply by viceice** [reply]
if you are using npm <v7 add `--no-install` to the npx call. Maybe it's time to create a [minimal reproduction](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) for futher help

**Reply by gjorgi93** [reply]
```
         {
           "lockFile": "code/js/frontend/package.json",
           "stderr": "Command failed: npm ci --ignore-scripts\nnpm ERR! code ENOENT\nnpm ERR! syscall open\nnpm ERR! path /tmp/renovate/repos/gitlab/dzoonix/test33/package.json\nnpm ERR! errno -2\nnpm ERR! enoent ENOENT: no such file or directory, open '/tmp/renovate/repos/gitlab/dzoonix/test33/package.json'\nnpm ERR! enoent This is related to npm not being able to find a file.\nnpm ERR! enoent \n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /home/ubuntu/.npm/_logs/2021-04-09T08_43_37_570Z-debug.log\n"
         }
```

```
         {
           "lockFile": "code/js/frontend/package.json",
           "stderr": "Command failed: npx --no-install ng update @angular/cli --from= --to=11.2.5 --migrateOnly --allowDirty --force\nnot found: ng\n"
         }
```

my postUpgradeTasks:

```
      "postUpgradeTasks": {
        "commands": [
          "npm ci --ignore-scripts",
          "npx --no-install ng update {{{depName}}} --from={{{currentVersion}}} --to={{{newVersion}}} --migrateOnly --allowDirty --force"
        ],
        "fileFilters": ["code/js/e2e/*", "code/js/frontend/*"]
      }
```

**Reply by gjorgi93** [reply]
Could it be connected with https://github.com/renovatebot/renovate/discussions/8274 ? Do i need to enter  {{packageFileDir}} before executing the command? I have more than one package.json in my repo

**Reply by rarkins** [reply]
Yes. But we must in list on a public reproduction if you need further help, because otherwise we're spending too much time guessing missing details.

**Reply by gjorgi93** [reply]
In case anyone need, this is what worked for me:

``` 
"cd {{packageFileDir}} && npm ci --ignore-scripts",
"cd {{packageFileDir}} && npx --no-install ng update --allow-dirty --force"
```

these two options `--migrateOnly --allowDirty` are deprecated, `--migrate-only --allow-dirty` are the correct ones, please update docs.

**Reply by viceice** [reply]
@HonkingGoose can you update the docs please?

**Reply by HonkingGoose** [reply]
> @HonkingGoose can you update the docs please?

See PR #9476. 😉
